code/game/objects/items/weapons/mop.dm

```diff
@@ -33,9 +40,31 @@
 			if(T)
 				T.clean(src, user)
 			user << SPAN_NOTICE("You have finished mopping!")
+	else
+		makeWet(A, user)
+
+
+/obj/item/weapon/mop/proc/makeWet(atom/A, mob/user)
+	for(var/container in get_wet_containers)
+		if(istype(A, container))
+			if(A.reagents)
+				if(!A.is_open_container())
+					user << SPAN_WARNING("\The [A] is closed!")
+					return
+				if(A.reagents.total_volume < 1)
```

Comment:
reagent volume is not an integer, it can be between 0 and 1.
Change this check to <= 0